### PR TITLE
create badge component 

### LINKED ISSUE #2861: (SP: 2) Create Badge component aligned with design system
#### Description:
Implement a reusable `Badge` component that meets the design system standards. This component will be used throughout the application to maintain consistency in badge styles, behavior, and accessibility.

[Figma](https://www.figma.com/design/liJZDYhUnDP15Pi9bipDcv/Space2Study-(Students)?node-id=21836-151311&t=ATTIq5lxdqHxDboc-4)

#### Requirements:
- **Styles:**
  - Follow design system variables for colors, typography, padding, and other styles.
  
- **Props:**
  - `variant`: Defines type (sm, lg).
  - `badgeContent`: Provide content that will be rendered in Badge.
  - `color`: Supports different colors.
  - `maxContent`: The max value that will be shown in the Badge.
  - `isVisible`: To show Badge.

Don't forget to cover component with tests and add to the storybook

### PR DESCRIPTION
**Pull request by nebby2105**
Created Badge component #2861

**Props:**

- `variant`: Determines the badge style (`sm` or `lg`).
- `color`: Supports `primary`, `success`, and `error` colors (default: `primary`).
- `isVisible`: Controls visibility of the badge (default: `true`).
- `badgeContent`: Numeric or textual content (applies to `lg `variant).
- `maxContent`: Specifies the maximum number to display, e.g., 99+ (default: `10`).
- `isZeroShown`: Toggles visibility of 0 content for the `lg` variant (default:`false`).

**Examples:**
- variant ='`lg`'
- badgeContent='`7`'
- rest props by default
Screenshot 2024-12-01 at 23 05 40

--------------------

- variant = '`sm`'
Screenshot 2024-12-01 at 23 05 48

--------------------

- color = '`success`'
Screenshot 2024-12-01 at 23 05 52

--------------------

- color='`error`'
Screenshot 2024-12-01 at 23 06 04

--------------------

- variant ='`lg`'
- color ='`success`'
- maxContent ='`99`'
Screenshot 2024-12-01 at 23 06 20

--------------------

- variant ='`lg`'
- badgeContent='`7`'
- color = '`error`'
Screenshot 2024-12-01 at 23 06 24

--------------------

- variant ='`lg`'
- badgeContent='`0`'
- isZeroShown
Screenshot 2024-12-01 at 23 06 39

--------------------

- [x] Implemented Badge component.
- [x] Added prop types and default values.
- [x] Integrated component into Storybook with examples.
- [x] Verified rendering and behavior through unit tests.

### CODE REVIEW

File: src/design-system/components/badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  isVisible?: boolean
+}
+
+type LargeBadgeProps = {
+  variant: 'lg'
+  badgeContent: number
+  maxContent?: number
+  color?: BadgeColor
+  isVisible?: boolean
+  showZero?: boolean
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/badge/Badge.tsx]: Name it as a boolean, please
```suggestion
  isZeroShown?: boolean
```

File: src/design-system/components/badge/Badge.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+type BadgeProps = (LargeBadgeProps | SmallBadgeProps) &
+  Omit<MuiBadgeProps, 'variant'>
+
+const Badge: FC<BadgeProps> = ({ children, ...props }) => {
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/badge/Badge.tsx]: Do not import FC from react, just use React.FC instead. It it more clean and readable
```suggestion
const Badge: React.FC<BadgeProps> = ({ children, ...props }) => {
```

File: src/design-system/components/badge/Badge.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  const displayBadge = (props.isVisible ?? true) ? props.badgeContent : 0
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/badge/Badge.tsx]: `isVisible` field is in both of your types (`SmallBadgeProps` and `LargeBadgeProps`). So you can make a destructurization of this field, like you did with `children`, and add a default value there.

File: src/design-system/components/badge/Badge.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  const badgeVariant = props.variant === 'sm' ? 'dot' : 'standard'
+  return (
+    <MuiBadge
+      badgeContent={displayBadge}
+      color={props.color ?? 'primary'}
+      data-testid='badge'
+      max={props.variant === 'lg' ? (props.maxContent ?? 10) : undefined}
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/badge/Badge.tsx]: Extract this expression into separate variabale, please, like you did with `badgeVariant`

File: src/design-system/components/badge/Badge.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  showZero?: boolean
+}
+
+type BadgeProps = (LargeBadgeProps | SmallBadgeProps) &
+  Omit<MuiBadgeProps, 'variant'>
+
+const Badge: FC<BadgeProps> = ({ children, ...props }) => {
+  const displayBadge = (props.isVisible ?? true) ? props.badgeContent : 0
+  const badgeVariant = props.variant === 'sm' ? 'dot' : 'standard'
+  return (
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/badge/Badge.tsx:37]: Add a blank line before return statement.
It is a really good practice to do that, but unfortunately we don't have a linter for that :(

File: src/design-system/components/badge/Badge.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    <MuiBadge
+      badgeContent={displayBadge}
+      color={props.color ?? 'primary'}
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/badge/Badge.tsx]: Same as with `isVisible`. Add default value with destructurization, please

**Review by ShadowOfTheSpace — CHANGES_REQUESTED** [review]

File: src/design-system/components/badge/Badge.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  children,
+  isVisible = true,
+  color = 'primary',
+  ...props
+}) => {
+  const displayBadge = isVisible ? props.badgeContent : 0
+  const badgeVariant = props.variant === 'sm' ? 'dot' : 'standard'
+  const maxContentShown =
+    props.variant === 'lg' ? (props.maxContent ?? 10) : undefined
+  const maxShown =
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/badge/Badge.tsx]:
```suggestion
  const isZeroShown =
```

File: src/design-system/components/badge/Badge.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+const Badge: React.FC<BadgeProps> = ({
+  children,
+  isVisible = true,
+  color = 'primary',
+  ...props
+}) => {
+  const displayBadge = isVisible ? props.badgeContent : 0
+  const badgeVariant = props.variant === 'sm' ? 'dot' : 'standard'
+  const maxContentShown =
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/badge/Badge.tsx]:
```suggestion
  const maxContent =
```

File: src/design-system/components/badge/Badge.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+type BadgeProps = (LargeBadgeProps | SmallBadgeProps) &
+  Omit<MuiBadgeProps, 'variant'>
+
+const Badge: React.FC<BadgeProps> = ({
+  children,
+  isVisible = true,
+  color = 'primary',
+  ...props
+}) => {
+  const displayBadge = isVisible ? props.badgeContent : 0
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/badge/Badge.tsx]:
```suggestion
  const badgeContent = isVisible ? props.badgeContent : 0
```

File: src/design-system/components/badge/Badge.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  const badgeVariant = props.variant === 'sm' ? 'dot' : 'standard'
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/badge/Badge.tsx]:
```suggestion
  const variant = props.variant === 'sm' ? 'dot' : 'standard'
```

**Review by ShadowOfTheSpace — CHANGES_REQUESTED** [review]
Just renaming)
You can keep names of variables like in props without any conflict. It would be much better and readable

**Review by ShadowOfTheSpace — APPROVED** [review]

**Review by amoutens — APPROVED** [review]

**Review by SofiiaYevush — APPROVED** [review]

**Pull request merged**